validate.go

```diff
@@ -16,36 +17,133 @@ func validate(payload []byte) ([]byte, error) {
 			kubewarden.Code(400))
 	}
 
-	logger.Info("validating request")
+	if len(settings.AllowedHostPaths) == 0 {
+		// empty settings, accepting
+		return kubewarden.AcceptRequest()
+	}
 
-	data := gjson.GetBytes(
+	volumes := gjson.GetBytes(
 		payload,
-		"request.object.metadata.name")
-
-	if !data.Exists() {
-		logger.Warn("cannot read object name from metadata: accepting request")
+		"request.object.spec.volumes")
+	if !volumes.Exists() {
+		// pod defines no volumes, accepting
 		return kubewarden.AcceptRequest()
 	}
-	name := data.String()
 
-	logger.DebugWithFields("validating ingress object", func(e onelog.Entry) {
-		namespace := gjson.GetBytes(payload, "request.object.metadata.namespace").String()
+	logger.DebugWithFields("validating pod object", func(e onelog.Entry) {
+		name := gjson.GetBytes(payload, "request.object.metadata.name").String()
+		namespace := gjson.GetBytes(payload,
+			"request.object.metadata.namespace").String()
 		e.String("name", name)
 		e.String("namespace", namespace)
 	})
 
-	if settings.DeniedNames.Contains(name) {
-		logger.InfoWithFields("rejecting ingress object", func(e onelog.Entry) {
+	// build list of volumeMounts from all the initContainers and containers:
+	volumeMounts := make([]gjson.Result, 0)
+	cases := []string{
+		"request.object.spec.initContainers",
+		"request.object.spec.containers",
+	}
+	for _, c := range cases {
+		containers := gjson.GetBytes(payload, c)
+		for _, container := range containers.Array() {
+			containerVolumeMounts := gjson.Get(container.String(),
+				"volumeMounts")
+			for _, mount := range containerVolumeMounts.Array() {
+				volumeMounts = append(volumeMounts, mount)
```

Comment:
Later in the next loop you skip the volume mounts that are not of type `hostPath`. Can't we move the check here and add to `volumeMounts` only the volumes that already satisfy this requirement?

Reply:
the `if` skips the `volumes` (not `volumeMounts`) that are not of type `hostPath`, so we can't take do it when building the list of `volumeMounts` (I got the same idea when writing it :D).

Reply:
I can reduce iterations by moving the if outside of the for though! done in https://github.com/kubewarden/psp-hostpaths-policy/pull/1/commits/27211f5bec75babc9b7887f9aa39077d74832d79.